napari/utils/colormaps/colormaps.py

```diff
@@ -273,57 +279,13 @@ def vispy_or_mpl_colormap(name):
                 'or matplotlib.'
             )
         mpl_colors = mpl_cmap(np.linspace(0, 1, 256))
-        cmap = Colormap(mpl_colors)
+        cmap = Colormap(name=name, colors=mpl_colors)
```

Comment:
perhaps have some `@classmethods` declared on our Colormap (`from_mpl` & `from_vispy`) ... so you can just do, e.g. `cmap = Colormap.from_mpl(mpl_cmap)`

Reply:
interesting idea, i'll look into that